client/lib/i18n-utils/utils.js

```diff
@@ -278,3 +278,81 @@ export function filterLanguageRevisions( languageRevisions ) {
 		return true;
 	} );
 }
+
+/**
+ * Potentially rewrite the timezone to what we have on the server-side.
+ * See https://core.trac.wordpress.org/ticket/26656
+ *
+ * @param {string} A timezone string, like Asia/Calcutta.
+ *
+ * @returns {string} A potentially rewritten timezone string, like Asia/Kolkata.
+ */
+
+export function maybeRewriteTimezone( timezone ) {
```

Comment:
Adding this function to the `lib/i18n-utils/utils` module inflates our entrypoint JS chunks by a few hundred bytes:
```
name                   parsed_size           gzip_size
entry-main                 +2260 B  (+0.1%)     +883 B  (+0.2%)
entry-login                +2260 B  (+0.2%)     +852 B  (+0.3%)
entry-domains-landing      +2260 B  (+0.4%)     +843 B  (+0.6%)
```
That's because other util functions from this module are used during Calypso boot and the whole module gets bundled in the entrypoint.

It would be better to create a new module for this function. E.g., `lib/i18n-utils/rewrite-timezone`.

Reply:
An even better abstraction would be a `lib/i18n-utils/guess-timezone` module that exports a function:
```js
export default guessTimezone() {
  return maybeRewriteTimezone( moment.tz.guess() );
}
```
That would be convenient for the `concierge/signup-form` reducer, too.

Reply:
Thanks for catching this, I have committed a new file. Let's see what iscfy says.